### PR TITLE
Add support for checklists

### PR DESCRIPTION
**Pull request by colebemis**
This PR introduces the concept of "checklists" on documentation pages, laying the groundwork for us to [add stable criteria checklists on component documentation pages](https://github.com/github/primer/issues/387).

| Frontmatter | Documentation page |
| --- | --- |
| CleanShot 2021-10-22 at 16 04 16@2x | CleanShot 2021-10-22 at 16 04 44@2x |

## Code review

I'm looking for feedback on the API and documentation of this new checklists feature. Does this abstraction feel "correct"?

## TODO

- [x] Write documentation
- [x] Write PR description
- [x] Create issues for follow up tasks
- [x] Implement checklists (based on documented API)
- [x] Validate checklis schemas during build (fail build if any schemas are invalid)
- [x] Validate frontmatter (fail build frontmatter is invalid)
- [x] Tests
- [x] Accessibility review of checklist items
- [x] Add a changeset

## Next steps (in follow up PRs)

This PR was getting a little too big so I decided to break it up. I'll create follow up PRs for these tasks:

- [ ] [Implement `groupBy` functionality](https://github.com/primer/doctocat/issues/281)
- [ ] [Add built-in component maturity checklist](https://github.com/primer/doctocat/issues/282)

### CODE REVIEW

File: docs/content/usage/checklists.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+- Examples
+- API reference
+
+We'll use a checklist to make these expectations visible.
+
+### Create a checklist schema
+
+First, we'll need to describe the structure of our checklist by creating a checklist schema. Checklist schema files (`.schema.js`) must live in a `src/@primer/gatsby-theme-doctocat/src/checklists/` directory. In our case, we'll create `documentation.schema.js`:
+
+<!-- NOTE: should schema files be in TypeSript? -->
```
[the remark this answers is not in the document]
Comment (colebemis) [reply docs/content/usage/checklists.md]: So it turns out that using typescript files in `gatsby-node.js` is not straightforward. I'm going to cut that out of the scope of this PR.

File: docs/content/usage/checklists.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+<!-- I'm still a little unsure about this API -->
+
+In our example scenerio, we now want to group checklist items by status. Specifically, we want to make it clear that title and description are required for all components with `Alpha` status. Examples and API reference documentation are required to move to `Beta` status.
+
+[show example of grouped checklist]
+
+To accomplish this we'll need to update our checklist schema:
+
+<!-- NOTE: `status` seems like the wrong name -->
```
[the remark this answers is not in the document]
Comment (colebemis) [reply docs/content/usage/checklists.md]: Will address this in a follow up PR

File: docs/content/usage/checklists.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+```diff
+// @primer/gatsby-theme-doctocat/src/checklists/accessibility.schema.js
+
+module.exports = {
+  title: 'Documentation checklist',
++ metadata: {
++  componentStatus: ['Alpha', 'Beta', 'Stable'],
++ },
++ groubBy: 'componentStatus',
```
[the remark this answers is not in the document]
Comment (colebemis) [reply docs/content/usage/checklists.md]: I'll implement groupBy in a follow up PR

File: theme/package.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
   "license": "MIT",
   "scripts": {
     "test": "jest",
-    "lint": "eslint '**/*.{js,ts,tsx}'",
+    "lint": "eslint '**/*.js'",
```
Comment (colebemis) [inline theme/package.json:8]: ESLint was failing on TS files. I'll follow up to fix this.

**Review by jfuchs — APPROVED** [review]
LGTM. It might be nice to use the canary release of this branch to set up an example checklist for primer react and take a look at _that_ PR and its canary deployment. WDYT?

**Comment by colebemis** [comment]
@jfuchs https://github.com/primer/react/pull/1589

**Review by alliethu — COMMENTED** [review]
The aria-labels look good to me. One thing I'm not sure about is if the checklist heading should be an `<h2>` by default.

**Pull request merged**